### PR TITLE
Liquid temp + split control for rad / fans + better USB handling + a few other improvements

### PR DESCRIPTION
**Pull request by goncalossilva**
I thought this repo was abandoned and kept improving levd in my own. This PR contains all changes I've made so far. Most commits are fairly granular, so I'd recommend reviewing the list to get an overview of all changes.

Aside from the README changes (ie. 5a6feb7), which makes sense to revert, I think all of these would be beneficial to the project.

### CODE REVIEW

**Comment by RBlafford** [comment]
This is great! I will take a look at this very soon

File: README.md [hunk 1]
```diff
@@ -136,13 +130,18 @@ $ sudo systemctl status levd.service
...
-better job then me :) 
+### Acknowledgements
+
+https://github.com/RBlafford/levd
+
+Most of the work is still [RBlafford](https://github.com/RBlafford)'s, on top of which I've added a number of improvements.
 
-This project would not have been possible without contributions to the open source community from
-jaksi.
+That repository contains aknowledgements of its own:
```
Comment (RBlafford) [inline README.md]: typo

File: kraken_driver.cpp [hunk 2]
```diff
@@ -26,8 +26,11 @@ KrakenDriver::KrakenDriver(libusb_device *kraken_device)
   // Grab endpoints via libusb
   CHECK (_config->bConfigurationValue == kMainConfigurationValue)
     << "bConfigurationValue must equal: " << kMainConfigurationValue;
-  CHECK(libusb_set_configuration(_handle, kMainConfigurationValue) == 0)
-    << "Error when setting kraken usb configuration";
+  int set_configuration_result =
```
Comment (RBlafford) [inline kraken_driver.cpp]: const int

File: leviathan_config.h [hunk 3]
```diff
@@ -15,18 +16,16 @@ struct Point {
...
   uint32_t main_color_{DEFAULT_RED};
-  uint32_t alt_color_{DEFAULT_RED};
 
-  bool enable_color_{false};
-  bool enable_blink_{false};
-
-  uint8_t blink_interval_{0};
-  uint8_t alter_interval_{0};
+  // Interval settings
+  uint32_t interval_{500};
```
Comment (RBlafford) [inline leviathan_config.h]: +1 I don't know why I left so many unused options in there.

File: README.md (hunk 1 above)
Comment (goncalossilva) [reply README.md]: I'd suggest simply reverting 5a6feb7 (ie. the README changes)? It doesn't make much sense if the changes will be merged back in.

Comment (RBlafford) [reply README.md]: You can just add a new commit to fix this , or you  can rebase the change and squash it into that commit with the README changes, then force push

File: leviathan_config.h [hunk 4]
```diff
@@ -15,18 +16,16 @@ struct Point {
 };
 
 struct leviathan_config {
-  // Fan profile
-  std::map<int32_t, LineFunction> ftp_;
+  // Fan/pump profile
+  std::string temp_source_{"cpu"};
```
Comment (RBlafford) [inline leviathan_config.h]: Use an "enum class" instead.

**Comment by RBlafford** [comment]
I like it! Small changes , then LGTM

**Comment by goncalossilva** [comment]
Thanks for the review! All feedback should be addressed, except for using an enum for cpu/liquid. I am unsure how to do it cleanly, considering the value is being parsed from a string in the configuration file (using yaml-cpp). I thought about using a map string->enum or macros, but both solutions seem overly complex for the purpose.

Please feel free to suggest a better approach or go in and make the changes you see fit.

**Comment by RBlafford** [comment]
Sure, i was thinking something like a function which converts the string to the enumclass called one time after parsing the yaml file. I can add it later, lets take these changes in asap

**Pull request merged**